File: chartiq/src/main/java/com/chartiq/sdk/scriptmanager/ScriptManager.java
package com.chartiq.sdk.scriptmanager;

import com.chartiq.sdk.model.AggregationType;
import com.chartiq.sdk.model.DrawingTool;
import com.chartiq.sdk.model.OHLCParams;

import org.json.JSONObject;

import java.util.Map;

public interface ScriptManager {

    String getDetermineOSScript();

    String getNativeQuoteFeedScript();

    String getAddDrawingListenerScript();

    String getAddLayoutListenerScript();

    String getSetSymbolScript(String symbol);

    // TODO: 26.08.20 Review the following method
    String getDateFromTickScript();

    String getSetDataMethodScript(String symbol);

    String getSetAccessibilityModeScript();

    String getIsChartAvailableScript();

    String getSetSymbolObjectScript(JSONObject symbolObject);

    String getSetPeriodicityScript(int period, String interval, String timeUnit);

    String getGetChartNameScript();

    String getPushDataScript(String symbol, OHLCParams[] data);

    String getPushUpdateScript(OHLCParams[] data);

    String getSetChartTypeScript(String chartType);

    String getAddComparisonScript(String symbol, String hexColor, Boolean isComparison);

    String getRemoveComparisonScript(String symbol);

    String getClearChartScript();

    String getSetChartScaleScript(String scale);

    String getAddStudyScript(String studyName, Map<String, Object> inputs, Map<String, Object> outputs, Map<String, Object> parameters);

    String getRemoveStudyScript(String studyName);

    String getRemoveAllStudiesScript();

    String getEnableCrosshairsScript();

    String getIsCrosshairsEnabledScript();

    String getDisableCrosshairsScript();

    String getGetCrosshairsHUDDetailScript();

    String getEnableDrawingScript(DrawingTool type);

    String getDisableDrawingScript();

    String getClearDrawingScript();

    String getSetDrawingParameterScript(String parameter, String value);

    String getSetStyleScript(String object, String parameter, String value);

    String getSetThemeScript(String theme);

    String getGetStudyListScript();

    String getGetActiveStudiesScript();

    String getSetAggregationTypeScript(AggregationType aggregationType);

    String getGetStudyInputParametersScript(String studyName);

    String getGetStudyOutputParametersScript(String studyName);

    String getGetStudyParametersScript(String studyName);

    String getSetStudyInputParameterScript(String studyName, String parameter, String value);

    String getSetStudyOutputParameterScript(String studyName, String parameter, String value);

    String getGetDrawingParametersScript(String drawingName);

    // TODO: 27.08.20 Review the following set of functions
    String getChangeChartStyleScript(final Object... args);

    String getSetChartPropertyScript(final String property, final Object value);

    String getGetChartPropertyScript(final String property);

    String getSetEnginePropertyScript(final String property, final Object value);

    String getGetEnginePropertyScript(final String property);

    String getParseDataScript(OHLCParams[] data, String callbackId);
}

File: chartiq/src/main/java/com/chartiq/sdk/scriptmanager/ScriptManager.kt
package com.chartiq.sdk.scriptmanager

import com.chartiq.sdk.model.ChartLayer
import com.chartiq.sdk.model.ChartTheme
import com.chartiq.sdk.model.OHLCParams
import com.chartiq.sdk.model.drawingtool.DrawingTool
import com.chartiq.sdk.model.study.StudyParameterModel

internal interface ScriptManager {

    fun getDetermineOSScript(): String

    fun getNativeQuoteFeedScript(): String

    fun getAddDrawingListenerScript(): String

    fun getAddLayoutListenerScript(): String

    fun getAddMeasureListener(): String

    fun getSetSymbolScript(symbol: String): String

    fun getDateFromTickScript(): String

    fun getLoadChartScript(): String

    fun getSetAccessibilityModeScript(): String

    fun getIsChartAvailableScript(): String

    fun getSetPeriodicityScript(period: Int, interval: String, timeUnit: String): String

    fun getPushDataScript(symbol: String, data: List<OHLCParams>): String

    fun getPushUpdateScript(data: List<OHLCParams>): String

    fun getSetChartTypeScript(chartType: String): String

    fun getChartTypeScript(): String

    fun getAggregationTypeScript(): String

    fun getResizeChartScript(): String

    fun getClearChartScript(): String

    fun getChartScaleScript(): String

    fun getSetChartScaleScript(scale: String): String

    fun getAddStudyScript(studyName: String): String

    fun getRemoveStudyScript(studyName: String): String

    fun getRemoveAllStudiesScript(): String

    fun getEnableCrosshairScript(value: Boolean): String

    fun getIsCrosshairsEnabledScript(): String

    fun getGetCrosshairHUDDetailsScript(): String

    fun getEnableDrawingScript(type: DrawingTool): String

    fun getDisableDrawingScript(): String

    fun getClearDrawingScript(): String

    fun getSetDrawingParameterScript(parameter: String, value: String): String

    fun getSetStyleScript(obj: String, parameter: String, value: String): String

    fun getSetThemeScript(theme: ChartTheme): String

    fun getGetStudyListScript(): String

    fun getGetActiveStudiesScript(): String

    fun getSetAggregationTypeScript(aggregationType: String): String

    fun getStudyInputParametersScript(studyName: String): String

    fun getStudyOutputParametersScript(studyName: String): String

    fun getStudyParametersScript(studyName: String): String

    fun getSetStudyParametersScript(name: String, parameters: List<StudyParameterModel>): String

    fun getSetStudyParameterScript(studyName: String, parameter: StudyParameterModel): String

    fun getGetDrawingParametersScript(drawingName: String): String

    fun getSetChartStyleScript(obj: String, attribute: String, value: String): String

    fun getSetChartPropertyScript(property: String, value: Any): String

    fun getGetChartPropertyScript(property: String): String

    fun getSetEnginePropertyScript(property: String, value: Any): String

    fun getGetEnginePropertyScript(property: String): String

    fun getParseDataScript(data: List<OHLCParams>, callbackId: String, moreAvailable: Boolean): String

    fun getInvertYAxisScript(): String

    fun getSetInvertYAxisScript(inverted: Boolean): String

    fun getIsExtendedHoursScript(): String

    fun getSetExtendedHoursScript(extended: Boolean): String

    fun getUndoDrawingScript(): String

    fun getRedoDrawingScript(): String

    fun getDeleteDrawingScript(): String

    fun getCloneDrawingScript(): String

    fun getLayerManagementScript(layer: ChartLayer): String

    fun getGetTranslationsScript(languageCode: String): String

    fun getSetLanguageScript(languageCode: String): String

    fun getRestoreDefaultDrawingConfigScript(tool: DrawingTool, all: Boolean): String

    fun getGetActiveSeriesScript(): String

    fun getAddSeriesScript(symbol: String, color: String, isComparison: Boolean): String

    fun getRemoveSeriesScript(symbol: String): String

    fun getSetSeriesParameterScript(symbol: String, field: String, value: String): String
}
